### DISCUSSION TITLE
Add support for `dagrun_failed` in REST API

### DISCUSSION THREAD

**Discussion by reganbaum** [apache/airflow#40602] [General]
### Description

Today, stopping the execution of a DAG run requires 2 REST API calls: one to identify non-completed tasks of a run, and another to mark those tasks as failed. This has the potential of running into race conditions, where tasks could complete between the two API calls, leading to tasks being marked as failed when they may have succeeded. Ideally, there is a single REST API that handles stopping the execution of a DAG run. The UI does this today by invoking the endpoint `/dagrun_failed`. The goal is to have a REST API endpoint available that stops a DAG run at `/api/v1/dagrun_failed`.

### Use case/motivation

I would like to programmatically stop a running workflow, which actually stops the execution (not just changes the state in the database).

### Related issues

https://github.com/apache/airflow/issues/40594 (The `dagrun_failed` issue was originally combined with the `dag_stats` issue, but separated it out into its own issue for clearer tracking)

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
As explained in #40594 It's a wish but if you want to have your wish to materialize, you need to explain and participate in the discussions on improving airflow architecture happening here:  https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-72+Task+Execution+Interface+aka+Task+SDK - and propose what kind of API and technology you would like to use there and why it is important. Until then it's not a "featiure" request, it's a discussion, because there is no clear way how this would work.

Converting it to a discussion.
